src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -12678,9 +12678,10 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
 
     flags = GetCompileFlags(ftn, flags, &methodInfo);
 
-    // If the reverse P/Invoke flag is used, we aren't going to support
-    // any tiered compilation.
-    if (flags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_REVERSE_PINVOKE))
+    // Clearing all tier flags and mark as optimized if the reverse P/Invoke
```

Comment:
Nit: Should this whole `if` be under `#ifdef FEATURE_TIERED_COMPILATION`?

Reply:
We could. The `IsEligibleForTieredCompilation()` isn't under that so I didn't think it was needed. The `IsEligibleForTieredCompilation` is hard coded outside the `#ifdef FEATURE_TIERED_COMPILATION` as `false` so the compiler should be able to remove it as dead code. What is your preference?

Reply:
Oh but the flag clearing isn't.... boo. Okay. I will remove it.